build-logic/src/main/kotlin/at/asitplus/gradle/SignumConventionsPlugin.kt

```diff
@@ -0,0 +1,332 @@
+import com.android.build.api.variant.KotlinMultiplatformAndroidComponentsExtension
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.tasks.StopExecutionException
+import org.gradle.api.tasks.testing.Test
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.kotlin.dsl.*
+import org.gradle.plugins.signing.SigningExtension
+import org.jetbrains.kotlin.gradle.ExperimentalWasmDsl
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+import org.jetbrains.kotlin.gradle.plugin.KotlinTarget
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget
+import org.jetbrains.kotlin.konan.target.Family
+import java.io.ByteArrayOutputStream
+
+/**
+ * Gradle convention plugin for Signum
+ */
+class SignumConventionsPlugin : Plugin<Project> {
+    override fun apply(target: Project) = with(target) {
+        logger.info("SignumC onventions Plugin applied to project: ${'$'}{target.path}")
+        pluginManager.apply("org.jetbrains.kotlin.multiplatform")
+        pluginManager.apply("org.jetbrains.kotlin.plugin.serialization")
+        pluginManager.apply("com.android.kotlin.multiplatform.library")
+        pluginManager.apply("signing")
+        pluginManager.apply("at.asitplus.gradle.conventions")
+        pluginManager.apply("de.infix.testBalloon")
+    }
+}
+
+class SignumConventionsExtension(private val project: Project) {
+    init {
+        val indispensableVersion: String by project.extra
+        project.version = indispensableVersion
+        //if we do this properly, cinterop (swift-klib) blows up, so we hack!
+        project.afterEvaluate {
+            tasks.withType<Test>().configureEach {
+                maxHeapSize = "4G"
+            }
+        }
+        project.silence()
+        project.fermentRottenApples()
+
+        project.extensions.getByType<KotlinMultiplatformExtension>().apply {
+            compilerOptions.freeCompilerArgs.add("-Xexpect-actual-classes")
+            sourceSets.whenObjectAdded {
+                languageSettings.optIn("kotlin.ExperimentalUnsignedTypes")
+            }
+        }
+
+        project.extensions.getByType<KotlinMultiplatformAndroidComponentsExtension>().apply {
+            onVariants { v ->
+                // Configure the instrumented-test APK only
+                v.androidTest?.manifestPlaceholders?.put("testLargeHeap", "true")
+            }
+        }
+    }
+
+    var supreme: Boolean = false
+        get() {
+            return field
+        }
+        set(value) {
+            if (!value) {
+                val indispensableVersion: String by project.extra
+                project.version = indispensableVersion
+            } else {
+                val supremeVersion: String by project.extra
+                project.version = supremeVersion
+                project.afterEvaluate {
```

Comment:
putting this in the setter of a boolean variable feels like a colossal hack; what if the variable is set to true, then false? i mean, yeah, it's our own code, but still...

is there any reason for this supreme-specific code to not be in the supreme build.gradle?